### PR TITLE
Fix/deprecations

### PR DESCRIPTION
**Pull request by azribilel**
- update composer requirments
- upgrade doctrine doctrine-behaviors
- fix deprecations 
- fix php docs
- create classe analayzer service

### CODE REVIEW

File: Resources/config/services.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ parameters:
   lch.media.pdf.regenerate_thumbnails.command.class: Lch\MediaBundle\Command\RegeneratePdfThumbnailsCommand
   lch.media.pdf.subscriber.class: Lch\MediaBundle\Listener\PdfSubscriber
 
+  lch.media.behavior.reflection.class_analyzer.class: Lch\MediaBundle\Behavior\ClassAnalyzer
```
Comment (devgiants) [inline Resources/config/services.yml]: The `Behavior` folder should only traits, no classes. Could you move it away please?

File: Resources/config/services.yml [hunk 2]
```diff
@@ -94,6 +96,10 @@ services:
     public: true
     lazy: true
 
+  lch.media.behavior.reflection.class_analyzer:
+    class: '%lch.media.behavior.reflection.class_analyzer.class%'
+    public: false
+
```
Comment (devgiants) [inline Resources/config/services.yml]: Same remark as above, this is not a behavior. You can rename it after moving it elsewhere

File: Validator/Constraints/HasAllowedFileExtensionValidator.php [hunk 3]
```diff
@@ -2,11 +2,9 @@
 
 namespace Lch\MediaBundle\Validator\Constraints;
 
-use Lch\MediaBundle\Behavior\Storable;
```
Comment (devgiants) [inline Validator/Constraints/HasAllowedFileExtensionValidator.php]: Why did this trait was removed?

File: Validator/Constraints/MediaWeightValidator.php [hunk 4]
```diff
@@ -2,12 +2,9 @@
 
 namespace Lch\MediaBundle\Validator\Constraints;
 
-use Lch\MediaBundle\Behavior\Storable;
```
Comment (devgiants) [inline Validator/Constraints/MediaWeightValidator.php]: Smae question as above, why did this trait was removed? I did'nt get the point

File: Resources/config/services.yml (hunk 1 above)
Comment (azribilel) [reply Resources/config/services.yml]: Ok, i will change it

File: Resources/config/services.yml (hunk 2 above)
Comment (azribilel) [reply Resources/config/services.yml]: Ok, i will change it

File: Validator/Constraints/HasAllowedFileExtensionValidator.php (hunk 3 above)
Comment (azribilel) [reply Validator/Constraints/HasAllowedFileExtensionValidator.php]: Sorry, i removed all unused classes, so i guess i remove with those classes.
I will return it

**Comment by azribilel** [comment]
I will close this and create a new clean pull request

**Pull request closed without merging**